<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Matrix scan. — matrix_scan • regutools</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Matrix scan. — matrix_scan" />
<meta property="og:description" content="Scan sequences with one or several position-specific scoring
matrices (PSSM)
to identify instances of the corresponding motifs (putative sites).
This function supports a variety of background models (Bernoulli, Markov
chains of any order)." />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">regutools</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/regutools.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ComunidadBioInfo/regutools">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Matrix scan.</h1>
    <small class="dont-index">Source: <a href='https://github.com/ComunidadBioInfo/regutools/blob/master/R/matrix_scan.R'><code>R/matrix_scan.R</code></a></small>
    <div class="hidden name"><code>matrix_scan.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Scan sequences with one or several position-specific scoring
matrices (PSSM)
to identify instances of the corresponding motifs (putative sites).
This function supports a variety of background models (Bernoulli, Markov
chains of any order).</p>
    </div>

    <pre class="usage"><span class='fu'>matrix_scan</span>(
  <span class='no'>sequence</span>,
  <span class='no'>matrix</span>,
  <span class='kw'>sequence.format</span> <span class='kw'>=</span> <span class='st'>"fasta"</span>,
  <span class='kw'>matrix.format</span> <span class='kw'>=</span> <span class='st'>"tab"</span>,
  <span class='kw'>quick</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>n.treatment</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>concensus.name</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>pseudo</span> <span class='kw'>=</span> <span class='fl'>1</span>,
  <span class='kw'>equi.pseudo</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>top.matrices</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>background.model</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>background</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>background.input</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>background.window</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>markov</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>background.pseudo</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>return.fields</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>sort.distrib</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>lth</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>uth</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>str</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>origin</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>decimals</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>crer.ids</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>organism</span> <span class='kw'>=</span> <span class='kw'>NULL</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>sequence</th>
      <td><p>Sequence(s) to scan - all the formats supported in RSAT can
be used as input (default: <code>fasta</code>)</p></td>
    </tr>
    <tr>
      <th>matrix</th>
      <td><p>Matrix/ces to scan with. The matrix format is specified with
the option <code>matrix.format</code> (see below) Default format: <code>tab</code>.</p></td>
    </tr>
    <tr>
      <th>sequence.format</th>
      <td><p>Supported fields:</p><ul>
<li><p><code>fasta</code> (default)</p></li>
<li><p><code>IG</code> (Intelligenetics)</p></li>
<li><p><code>WC</code> (wconsensus)</p></li>
<li><p><code>raw</code></p></li>
</ul></td>
    </tr>
    <tr>
      <th>matrix.format</th>
      <td><p>Supported fields:</p><ul>
<li><p><code>tab</code> (default)</p></li>
<li><p><code>cb</code></p></li>
<li><p><code>consensus</code></p></li>
<li><p><code>gibbs</code></p></li>
<li><p><code>meme</code></p></li>
<li><p><code>assembly</code></p></li>
</ul></td>
    </tr>
    <tr>
      <th>quick</th>
      <td><p>Delegates scanning to the C program matrix-scan-quick
(developed by Matthieu Defrance). Evaluate if the quick mode is compatible
with the selected output parameters, otherwise, run in the slower mode.
Incompatible with - CRER scanning - window background model.</p></td>
    </tr>
    <tr>
      <th>n.treatment</th>
      <td><p>Treatment of N characters. These characters are often
used in DNA sequences to represent undefined or masked nucleotides.</p><ul>
<li><p><code>skip</code>.  N-containing regions are skipped.</p></li>
<li><p><code>score</code>. N-containing regions are scored. The probability of an N is 1
for both the background model and the matrix. The N residues will thus
contribute neither positively nor negatively to the weight score of the
N-containing fragment. This option can be useful to detect sites which are
at the border of N-containing regions, or in cases there are isolated N in
the sequences.</p></li>
</ul></td>
    </tr>
    <tr>
      <th>pseudo</th>
      <td><p>Pseudo-count for the matrix (default: 1). The pseudo-count
reflects the possibility that residues that were not (yet) observed in the
model might however be valid for future observations. The pseudo-count is
used to compute the corrected residue frequencies.</p></td>
    </tr>
    <tr>
      <th>equi.pseudo</th>
      <td><p>If this option is called,  the pseudo-weight is
distributed in an equiprobable way between residues. By default, the
pseudo-weight is distributed proportionally to residue priors, except for
the -window option where equipseudo is default.</p></td>
    </tr>
    <tr>
      <th>top.matrices</th>
      <td><p>Only scan with the top # matrices per matrix file. This
option is valid for some file formats containing multiple matrices where top
matrices are generally more informative.</p></td>
    </tr>
    <tr>
      <th>background.model</th>
      <td><p>Background model is a tab-delimited specification of
oligonucleotide frequencies.</p></td>
    </tr>
    <tr>
      <th>background</th>
      <td><p>To use a precalculated background model from RSAT. Works
with organism and markov options. Type of sequences used as background model
for estimating expected oligonucleotide frequencies.
Supported:</p><ul>
<li><p><code>upstream</code></p></li>
<li><p><code>upstream-noorf</code></p></li>
<li><p><code>upstream-noorf-rm</code></p></li>
<li><p><code>intergenic</code></p></li>
</ul></td>
    </tr>
    <tr>
      <th>background.input</th>
      <td><p>Calculate background model from the input sequence
set. This option requires to specify the order of the background model with
the option markov.</p></td>
    </tr>
    <tr>
      <th>background.window</th>
      <td><p>Size of the sliding window for the background model
calculation. This option requires to specify the order of the background
model with the option markov (suitable for short order model only markov 0
or 1)</p></td>
    </tr>
    <tr>
      <th>markov</th>
      <td><p>Order of the markov chain for the background model. This
option is incompatible with the option background.</p></td>
    </tr>
    <tr>
      <th>background.pseudo</th>
      <td><p>Pseudo frequency for the background models. Value
must be a real between 0 and 1.
If this option is not specified,  the pseudo-frequency value depends on the
background calculation.
For <code>background.input</code> and <code>background_window</code>,  the pseudo frequency is
automatically calculated with the length (L) of the sequence following this
formula:
square-root of L divided by L+squareroot of L.
For <code>background.model</code>,  default value is 0.01.
If the training sequence length (L) is known,  the value can be set by
<code>background.pseudo</code> option to square-root of L divided by L+squareroot of L.</p></td>
    </tr>
    <tr>
      <th>return.fields</th>
      <td><p>List of fields to return.
Supported fields:</p><ul>
<li><p><code>sites</code></p></li>
<li><p><code>rank</code></p></li>
<li><p><code>limits</code></p></li>
<li><p><code>normw</code></p></li>
<li><p><code>bg_model</code></p></li>
<li><p><code>matrix</code></p></li>
<li><p><code>freq_matrix</code></p></li>
<li><p><code>weight_matrix</code></p></li>
<li><p><code>distrib</code></p></li>
</ul></td>
    </tr>
    <tr>
      <th>sort.distrib</th>
      <td><p>Sort score distribution by decreasing value of
significance, if value TRUE. By default, the score distributions are sorted
by score (weight).</p></td>
    </tr>
    <tr>
      <th>lth</th>
      <td><p>Lower threshold on some parameter. Format=list of
'parameter value'.
Supported fields:</p><ul>
<li><p><code>score</code></p></li>
<li><p><code>pval</code></p></li>
<li><p><code>eval</code></p></li>
<li><p><code>sig</code></p></li>
<li><p><code>normw</code></p></li>
<li><p><code>proba_M</code></p></li>
<li><p><code>proba_B</code></p></li>
<li><p><code>rank</code></p></li>
<li><p><code>crer_sites</code></p></li>
<li><p><code>crer_size</code></p></li>
<li><p><code>occ</code></p></li>
<li><p><code>occ_sum</code></p></li>
<li><p><code>inv_cum</code></p></li>
<li><p><code>exp_occ</code></p></li>
<li><p><code>occ_pval</code></p></li>
<li><p><code>occ_eval</code></p></li>
<li><p><code>occ_sig</code></p></li>
<li><p><code>occ_sig_rank</code></p></li>
</ul></td>
    </tr>
    <tr>
      <th>uth</th>
      <td><p>Upper threshold on some parameter. Format=list of 'param value'.
Supported parameters: same as <code>lth</code>.</p></td>
    </tr>
    <tr>
      <th>str</th>
      <td><p>Scan 1 or 2 strands for DNA sequences.</p></td>
    </tr>
    <tr>
      <th>origin</th>
      <td><p>Define the origin for the calculation of positions.
<code>origin</code> -0 defines the end of each sequence as the origin.
The matching positions are then negative values, providing the distance
between the match and the end of the sequence.</p></td>
    </tr>
    <tr>
      <th>decimals</th>
      <td><p>Number of decimals displayed for the weight score.</p></td>
    </tr>
    <tr>
      <th>crer.ids</th>
      <td><p>Assign one separate feature ID per CRER. This option is
convenient to distinguish separate CRERs.</p></td>
    </tr>
    <tr>
      <th>organism</th>
      <td><p>To use a precalculated background model from RSAT, choose
the organism corresponding to the background model. Works with background
and markov options.</p></td>
    </tr>
    <tr>
      <th>consensus.name</th>
      <td><p>Use the motif (degenerate) consensus as matrix name.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>an R object with results retrieved from RSAT</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='co'>## Define the matrix</span>
<span class='no'>m</span> <span class='kw'>&lt;-</span>
    <span class='st'>'; MET4 matrix, from Gonze et al. (2005). Bioinformatics 21, 3490-500.
A |   7   9   0   0  16   0   1   0   0  11   6   9   6   1   8
C |   5   1   4  16   0  15   0   0   0   3   5   5   0   2   0
G |   4   4   1   0   0   0  15   0  16   0   3   0   0   2   0
T |   0   2  11   0   0   1   0  16   0   2   2   2  10  11   8
//
; MET31 matrix, from Gonze et al. (2005). Bioinformatics 21, 3490-500.
A |   3   6   9   6  14  18  16  18   2   0   0   0   1   3   8
C |   8   3   3   2   3   0   1   0  13   2   0   1   0   3   6
G |   4   3   4   8   0   0   1   0   2   0  17   1  17  11   1
T |   3   6   2   2   1   0   0   0   1  16   1  16   0   1   3
//
; PHO4 matrix, from Gonze et al. (2005). Bioinformatics 21, 3490-500.
A |   0   4   4   1   1  21   0   0   0   0   2   2   6   1   7
C |   2   7  12   6  20   0  20   0   1   0   5   5   8   4   6
G |   5   1   2  11   0   0   0  21   0  15   8   7   2  11   2
T |  14   9   3   3   0   0   1   0  20   6   6   7   5   5   6
'</span>

<span class='co'>## This is incomplete:</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      <li><a href="#value">Value</a></li>
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>José Alquicira Hernández</p>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Joselyn Chavez, Carmina Barberena-Jonas, Jesus E. Sotelo-Fonseca, Jose Alquicira-Hernandez, Heladia Salgado, Alejandro Reyes, Leonardo Collado-Torres.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


