<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>regutools: an R package for the extraction of gene regulatory networks from RegulonDB • regutools</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="regutools: an R package for the extraction of gene regulatory networks from RegulonDB">
<meta property="og:description" content="regutools">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">regutools</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/regutools.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ComunidadBioInfo/regutools/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="regutools_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>regutools: an R package for the extraction of gene regulatory networks from RegulonDB</h1>
                        <h4 class="author">Joselyn Chávez</h4>
            <address class="author_afil">
      Instituto de Biotecnología, Universidad Nacional Autónoma de México, Cuernavaca, Morelos, MX<br><a class="author_email" href="mailto:#"></a><a href="mailto:joselynchavezf@gmail.com" class="email">joselynchavezf@gmail.com</a>
      </address>
                              <h4 class="author">Carmina Barberena Jonas</h4>
            <address class="author_afil">
      LANGEBIO, Cinvestav, Irapuato, Guanajuato, MX<br><a class="author_email" href="mailto:#"></a><a href="mailto:car.barjon@gmail.com" class="email">car.barjon@gmail.com</a>
      </address>
                              <h4 class="author">Jesus Emiliano Sotelo-Fonseca</h4>
            <address class="author_afil">
      LANGEBIO, Cinvestav, Irapuato, Guanajuato, MX<br><a class="author_email" href="mailto:#"></a><a href="mailto:jemiliano@gmail.com" class="email">jemiliano@gmail.com</a>
      </address>
                              <h4 class="author">Jose Alquicira Hernandez</h4>
            <address class="author_afil">
      Centro de Ciencias Genómicas, Universidad Nacional Autónoma de México, Cuernavaca, Morelos, MX<br><h4 class="author">Heladia Salgado</h4>
            <address class="author_afil">
      Centro de Ciencias Genómicas, Universidad Nacional Autónoma de México, Cuernavaca, Morelos, MX<br><a class="author_email" href="mailto:#"></a><a href="mailto:heladia@ccg.unam.mx" class="email">heladia@ccg.unam.mx</a>
      </address>
                              <h4 class="author">Leonardo Collado-Torres</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical Campus, Baltimore, MD, US<br><a class="author_email" href="mailto:#"></a><a href="mailto:lcolladotor@gmail.com" class="email">lcolladotor@gmail.com</a>
      </address>
                              <h4 class="author">Alejandro Reyes</h4>
            <address class="author_afil">
      Dana Farber Cancer Institute, Boston, MA, US<br><a class="author_email" href="mailto:#"></a><a href="mailto:alejandro.reyes.ds@gmail.com" class="email">alejandro.reyes.ds@gmail.com</a>
      </address>
                  
            <h4 class="date">11 October 2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ComunidadBioInfo/regutools/blob/master/vignettes/regutools.Rmd"><code>vignettes/regutools.Rmd</code></a></small>
      <div class="hidden name"><code>regutools.Rmd</code></div>

    </address>
</div>

    
    
<div id="basics" class="section level1">
<h1 class="hasAnchor">
<a href="#basics" class="anchor"></a>Basics</h1>
<div id="install-regutools" class="section level2">
<h2 class="hasAnchor">
<a href="#install-regutools" class="anchor"></a>Install <code>regutools</code>
</h2>
<p><code>R</code> is an open-source statistical environment which can be easily modified to enhance its functionality via packages. <em><a href="https://bioconductor.org/packages/3.12/regutools">regutools</a></em> is a <code>R</code> package available via Bioconductor. <code>R</code> can be installed on any operating system from <a href="https://cran.r-project.org/">CRAN</a> after which you can install <em><a href="https://bioconductor.org/packages/3.12/regutools">regutools</a></em> by using the following commands in your <code>R</code> session:</p>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
      <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
  <span class="op">}</span>

<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"regutools"</span><span class="op">)</span>

<span class="co">## Check that you have a valid Bioconductor installation</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/valid.html">valid</a></span><span class="op">(</span><span class="op">)</span></pre></div>
</div>
<div id="required-knowledge" class="section level2">
<h2 class="hasAnchor">
<a href="#required-knowledge" class="anchor"></a>Required knowledge</h2>
<p><em><a href="https://bioconductor.org/packages/3.12/regutools">regutools</a></em> is based on many other packages and in particular in those that have implemented the infrastructure needed for dealing with genomic and sequence data. That is, packages like <em><a href="https://bioconductor.org/packages/3.12/Biostrings">Biostrings</a></em> that allow you to work with sequences and <em><a href="https://bioconductor.org/packages/3.12/GenomicRanges">GenomicRanges</a></em> for data on genomic coordinates. A <em><a href="https://bioconductor.org/packages/3.12/regutools">regutools</a></em> user is not expected to deal with those packages directly but will need to be familiar with them to understand the results <em><a href="https://bioconductor.org/packages/3.12/regutools">regutools</a></em> generates. Furthermore, it’ll be useful for the user to know the syntax of <em><a href="https://bioconductor.org/packages/3.12/AnnotationHub">AnnotationHub</a></em> <a id="cite-Morgan_2020"></a>(<a href="#bib-Morgan_2020">Morgan and Shepherd, 2020</a>) in order to query and load the data provided by this package.</p>
<p>If you are asking yourself the question “Where do I start using Bioconductor?” you might be interested in <a href="http://lcolladotor.github.io/2014/10/16/startbioc/#.VkOKbq6rRuU">this blog post</a>.</p>
</div>
<div id="asking-for-help" class="section level2">
<h2 class="hasAnchor">
<a href="#asking-for-help" class="anchor"></a>Asking for help</h2>
<p>As package developers, we try to explain clearly how to use our packages and in which order to use the functions. But <code>R</code> and <code>Bioconductor</code> have a steep learning curve so it is critical to learn where to ask for help. The blog post quoted above mentions some but we would like to highlight the <a href="https://support.bioconductor.org/">Bioconductor support site</a> as the main resource for getting help regarding Bioconductor. Other alternatives are available such as creating GitHub issues and tweeting. However, please note that if you want to receive help you should adhere to the <a href="http://www.bioconductor.org/help/support/posting-guide/">posting guidelines</a>. It is particularly critical that you provide a small reproducible example and your session information so package developers can track down the source of the error.</p>
</div>
<div id="citing-regutools" class="section level2">
<h2 class="hasAnchor">
<a href="#citing-regutools" class="anchor"></a>Citing <code>regutools</code>
</h2>
<p>We hope that <em><a href="https://bioconductor.org/packages/3.12/regutools">regutools</a></em> will be useful for your research. Please use the following information to cite the package and the overall approach. Thank you!</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="co">## Citation info</span>
<span class="fu"><a href="https://rdrr.io/r/utils/citation.html">citation</a></span><span class="op">(</span><span class="st">"regutools"</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Chávez J, Barberena-Jonas C, Sotelo-Fonseca JE, Alquicira-Hernandez J,</span>
<span class="co">#&gt; Salgado H, Collado-Torres L, Reyes A (2020). _regutools: an R package</span>
<span class="co">#&gt; for data extraction from RegulonDB_. doi: 10.18129/B9.bioc.regutools</span>
<span class="co">#&gt; (URL: https://doi.org/10.18129/B9.bioc.regutools),</span>
<span class="co">#&gt; https://github.com/comunidadbioinfo/regutools - R package version</span>
<span class="co">#&gt; 1.1.3, &lt;URL: http://www.bioconductor.org/packages/regutools&gt;.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Chávez J, Barberena-Jonas C, Sotelo-Fonseca JE, Alquicira-Hernandez J,</span>
<span class="co">#&gt; Salgado H, Collado-Torres L, Reyes A (2020). "Programmatic access to</span>
<span class="co">#&gt; bacterial regulatory networks with regutools." _Bioinformatics_. doi:</span>
<span class="co">#&gt; 10.1093/bioinformatics/btaa575 (URL:</span>
<span class="co">#&gt; https://doi.org/10.1093/bioinformatics/btaa575), &lt;URL:</span>
<span class="co">#&gt; https://academic.oup.com/bioinformatics/advance-article-abstract/doi/10.1093/bioinformatics/btaa575/5861528&gt;.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; To see these entries in BibTeX format, use 'print(&lt;citation&gt;,</span>
<span class="co">#&gt; bibtex=TRUE)', 'toBibtex(.)', or set</span>
<span class="co">#&gt; 'options(citation.bibtex.max=999)'.</span></pre></div>
</div>
</div>
<div id="overview" class="section level1">
<h1 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h1>
<p><em>Escherichia coli K-12 (E. coli)</em> is the best bacterial organism studied to date. Thousands of papers have been published using <em>E. coli</em> as a model system asking how genes are regulated. The throughput of these experiments range from single-gene studies to whole-genome approaches. Twenty years ago, the database <a href="http://regulondb.ccg.unam.mx/"><em>RegulonDB</em></a> started collecting, curating and organizing this information into a centralized resource. Thanks to this huge efforts, researchers have had an easy way to access all these data in a database, facilitating the advancements of complete fields, such as <em>systems biology</em>.</p>
<p>The analysis of high-throughput experiments -such as RNA-seq or ChIP-seq- often requires the integration of databases such as <em>RegulonDB</em> in order to give biological interpretations to these data. The <em>regutools</em> package is designed to facilitate such integration by providing programmatic access to <em>RegulonDB</em> within the R environment <a id="cite-Chavez_2020"></a>(<a href="https://academic.oup.com/bioinformatics/advance-article-abstract/doi/10.1093/bioinformatics/btaa575/5861528">Chávez, Barberena-Jonas, Sotelo-Fonseca, Alquicira-Hernandez, et al., 2020</a>). The package retrieves information from the <em>RegulonDB</em> database into <em>Bioconductor</em> objects, ready for downstream analyses.</p>
<p>The package defines the object <strong>regulondb</strong>, which is a data structure that contains the path to a <em>SQLite</em> database retrieved from <em>RegulonDB</em> along with metadata such as database version and reference genome. The function <code><a href="../reference/connect_database.html">connect_database()</a></code> will retrieve the latest version of the database and connect to it. It will download the database using <em><a href="https://bioconductor.org/packages/3.12/AnnotationHub">AnnotationHub</a></em> or a backup mechanism if necessary. The <em>regutools</em> package contains functions with the most popular queries to <em>regutools</em>, such as retrieving information of which gene targets are regulated by a transcription factor. But users can also design queries that are specific to their analyses. This vignette describes how to use the provided functions and how to design programmatic queries to <em>regutools</em>. The general syntax of the function calls of this package is <code>result &lt;- functionCalled( regulondb, arguments )</code>.</p>
</div>
<div id="the-regulondb-object" class="section level1">
<h1 class="hasAnchor">
<a href="#the-regulondb-object" class="anchor"></a>The <strong>regulondb</strong> object</h1>
<p>The <strong>regulondb</strong> object is an extension of an <strong>SQLiteConnection</strong> class that host a connection to a database with the table structure defined in the <em>RegulonDB</em> database. It contains additional slots that specify the organism, genome version and database version. The function <code><a href="../reference/regulondb.html">regulondb()</a></code> is the constructor function of <strong>regulondb</strong> objects. This function receives as input a file path to the database file as well as information about the annotation as character vectors.</p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/ComunidadBioInfo/regutools">"regutools"</a></span><span class="op">)</span>

<span class="co">## Other packages used</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"Biostrings"</span><span class="op">)</span>

<span class="co">## Connect to the RegulonDB database</span>
<span class="va">regulondb_conn</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/connect_database.html">connect_database</a></span><span class="op">(</span><span class="op">)</span>

<span class="co">## Build a regulondb object</span>
<span class="va">e_coli_regulondb</span> <span class="op">&lt;-</span>
    <span class="fu"><a href="../reference/regulondb.html">regulondb</a></span><span class="op">(</span>
        database_conn <span class="op">=</span> <span class="va">regulondb_conn</span>,
        organism <span class="op">=</span> <span class="st">"E.coli"</span>,
        database_version <span class="op">=</span> <span class="st">"1"</span>,
        genome_version <span class="op">=</span> <span class="st">"1"</span>
    <span class="op">)</span>

<span class="va">e_coli_regulondb</span>
<span class="co">#&gt; regulondb object</span>
<span class="co">#&gt;   organism: E.coli</span>
<span class="co">#&gt;   genome_build: 1</span>
<span class="co">#&gt;   database_version: 1</span>
<span class="co">#&gt;   database_conn: /tmp/RtmpwjFJuz/BiocFileCache/b7d4d0c7b45_regulondb_v10.6.2_DM_sqlite3.db%3Fdl%3D1</span></pre></div>
<p>In order to get an overview of the tables present in a <em>regulondb</em> object, we can use the function <code><a href="../reference/list_datasets.html">list_datasets()</a></code>. This function will output all the available tables (datasets) that can be used to build queries.</p>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="fu"><a href="../reference/list_datasets.html">list_datasets</a></span><span class="op">(</span><span class="va">e_coli_regulondb</span><span class="op">)</span>
<span class="co">#&gt; [1] "DNA_OBJECTS"       "GENE"              "NETWORK"          </span>
<span class="co">#&gt; [4] "OPERON"            "PROMOTER"          "REGULONDB_OBJECTS"</span>
<span class="co">#&gt; [7] "TF"                "TU"</span></pre></div>
<p>For each table in the database, users can explore the fields (or attributes) of each table using the function <code>list_attributes</code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/list_attributes.html">list_attributes</a></span><span class="op">(</span><span class="va">e_coli_regulondb</span>, <span class="st">"GENE"</span><span class="op">)</span>, <span class="fl">8</span><span class="op">)</span>
<span class="co">#&gt; [1] "id"       "name"     "bnumber"  "gi"       "synonyms" "posleft"  "posright"</span>
<span class="co">#&gt; [8] "strand"</span></pre></div>
</div>
<div id="retrieving-data" class="section level1">
<h1 class="hasAnchor">
<a href="#retrieving-data" class="anchor"></a>Retrieving data</h1>
<p>Since the <em>regulondb</em> object is an extension of the <em>SQLiteConnection</em>, users can retrieve data from a <em>regulondb</em> object using the function <code>dbGetQuery()</code>. Additionally, this package provides a wrapper function to build queries to the database. This function is called <code><a href="../reference/get_dataset.html">get_dataset()</a></code> and has a very similar syntax to the <code>getBM()</code> function from the biomaRt package. The main arguments of the <code><a href="../reference/get_dataset.html">get_dataset()</a></code> function are a <em>regulondb</em> object, a dataset (or table) of the database, the fields of the dataset to retrieve (attributes) and filters to specify what information to get. The code below shows an example where three attributes from the dataset <strong>“GENE”</strong> for the genes <em>araC</em>, <em>crp</em> and <em>lacI</em>. Note that if the <code>filters=</code> parameter is empty, the function will retrieve all the hits it find in the database.</p>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="fu"><a href="../reference/get_dataset.html">get_dataset</a></span><span class="op">(</span>
    regulondb <span class="op">=</span> <span class="va">e_coli_regulondb</span>,
    dataset <span class="op">=</span> <span class="st">"GENE"</span>,
    attributes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"posleft"</span>, <span class="st">"posright"</span>, <span class="st">"strand"</span>, <span class="st">"name"</span><span class="op">)</span>,
    filters <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="st">"name"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"araC"</span>, <span class="st">"crp"</span>, <span class="st">"lacI"</span><span class="op">)</span><span class="op">)</span>
<span class="op">)</span>
<span class="co">#&gt; regulondb_result with 3 rows and 4 columns</span>
<span class="co">#&gt;     posleft  posright      strand        name</span>
<span class="co">#&gt;   &lt;integer&gt; &lt;integer&gt; &lt;character&gt; &lt;character&gt;</span>
<span class="co">#&gt; 1     70387     71265     forward        araC</span>
<span class="co">#&gt; 2   3486120   3486752     forward         crp</span>
<span class="co">#&gt; 3    366428    367510     reverse        lacI</span></pre></div>
<p>Some of the filters, such as <em>posright</em> or <em>posleft</em>, can be filtered by specifying intervals. For example, the code below indicates that all the start positions *posright" should be between position 1 and position 5000 of the genome. The parameter <code>inverval=</code> is used to specify that the filter for that field will be defined by an interval rather than a exact match.</p>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="fu"><a href="../reference/get_dataset.html">get_dataset</a></span><span class="op">(</span>
    <span class="va">e_coli_regulondb</span>,
    attributes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"posright"</span>, <span class="st">"name"</span><span class="op">)</span>,
    filters <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="st">"posright"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">5000</span><span class="op">)</span><span class="op">)</span>,
    interval <span class="op">=</span> <span class="st">"posright"</span>,
    dataset <span class="op">=</span> <span class="st">"GENE"</span>
<span class="op">)</span>
<span class="co">#&gt; regulondb_result with 3 rows and 2 columns</span>
<span class="co">#&gt;    posright        name</span>
<span class="co">#&gt;   &lt;integer&gt; &lt;character&gt;</span>
<span class="co">#&gt; 1      2799        thrA</span>
<span class="co">#&gt; 2      3733        thrB</span>
<span class="co">#&gt; 3       255        thrL</span></pre></div>
</div>
<div id="the-regulondb_result-object-and-integration-into-the-bioc-ecosystem" class="section level1">
<h1 class="hasAnchor">
<a href="#the-regulondb_result-object-and-integration-into-the-bioc-ecosystem" class="anchor"></a>The <strong>regulondb_result</strong> object and integration into the <em>BioC</em> ecosystem</h1>
<p>By default, the function <code><a href="../reference/get_dataset.html">get_dataset()</a></code> outputs a <strong>regulondb_result</strong> object, which is an extension of a <strong>DataFrame</strong> that stores information about the query used to generate this object. This additional information includes the organism name, the database and genome versions, and the table (or dataset) of the <em>regulondb</em> object that was queried by <code><a href="../reference/get_dataset.html">get_dataset()</a></code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit">
<span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_dataset.html">get_dataset</a></span><span class="op">(</span>
    regulondb <span class="op">=</span> <span class="va">e_coli_regulondb</span>,
    dataset <span class="op">=</span> <span class="st">"GENE"</span>,
    attributes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"posleft"</span>, <span class="st">"posright"</span>, <span class="st">"strand"</span>, <span class="st">"name"</span><span class="op">)</span>,
    filters <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="st">"name"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"araC"</span>, <span class="st">"crp"</span>, <span class="st">"lacI"</span><span class="op">)</span><span class="op">)</span>
<span class="op">)</span>
<span class="fu">slotNames</span><span class="op">(</span><span class="va">res</span><span class="op">)</span>
<span class="co">#&gt;  [1] "organism"         "genome_version"   "database_version" "dataset"         </span>
<span class="co">#&gt;  [5] "rownames"         "nrows"            "listData"         "elementType"     </span>
<span class="co">#&gt;  [9] "elementMetadata"  "metadata"</span></pre></div>
<p>To enable integration with other Bioconductor packages, we provide the function <code><a href="../reference/convert_to_granges.html">convert_to_granges()</a></code> which converts a <strong>regulondb_result</strong> object into a <strong>GRanges</strong> object whenever possible. For example, the result stored in the variable <code>res</code> has genomic coordinates and it is thus possible convert <code>res</code> into a <strong>GRanges</strong> object.</p>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="fu"><a href="../reference/convert_to_granges.html">convert_to_granges</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span>
<span class="co">#&gt; GRanges object with 3 ranges and 1 metadata column:</span>
<span class="co">#&gt;       seqnames          ranges strand |        name</span>
<span class="co">#&gt;          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span>
<span class="co">#&gt;   [1]   E.coli     70387-71265      + |        araC</span>
<span class="co">#&gt;   [2]   E.coli 3486120-3486752      + |         crp</span>
<span class="co">#&gt;   [3]   E.coli   366428-367510      - |        lacI</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></pre></div>
<p>An alternative way to get to the same result is to use the parameter <code>output_format=</code> directly in the function <code><a href="../reference/get_dataset.html">get_dataset()</a></code>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="fu"><a href="../reference/get_dataset.html">get_dataset</a></span><span class="op">(</span>
    regulondb <span class="op">=</span> <span class="va">e_coli_regulondb</span>,
    dataset <span class="op">=</span> <span class="st">"GENE"</span>,
    attributes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"posleft"</span>, <span class="st">"posright"</span>, <span class="st">"strand"</span>, <span class="st">"name"</span><span class="op">)</span>,
    filters <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="st">"name"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"araC"</span>, <span class="st">"crp"</span>, <span class="st">"lacI"</span><span class="op">)</span><span class="op">)</span>,
    output_format <span class="op">=</span> <span class="st">"GRanges"</span>
<span class="op">)</span>
<span class="co">#&gt; GRanges object with 3 ranges and 1 metadata column:</span>
<span class="co">#&gt;       seqnames          ranges strand |        name</span>
<span class="co">#&gt;          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span>
<span class="co">#&gt;   [1]   E.coli     70387-71265      + |        araC</span>
<span class="co">#&gt;   [2]   E.coli 3486120-3486752      + |         crp</span>
<span class="co">#&gt;   [3]   E.coli   366428-367510      - |        lacI</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></pre></div>
<p>In a similar manner, the function <code><a href="../reference/convert_to_biostrings.html">convert_to_biostrings()</a></code> converts <strong>regulondb</strong> objects into objects from the <em><a href="https://bioconductor.org/packages/3.12/Biostrings">Biostrings</a></em> package. Possible outputs of <code><a href="../reference/convert_to_biostrings.html">convert_to_biostrings()</a></code> are <strong>DNAStringSet</strong> objects if <code>seq_type="DNA"</code> or a <strong>BStringSet</strong> if <code>seq_type="product"</code>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="va">res_dnastring</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_dataset.html">get_dataset</a></span><span class="op">(</span>
    regulondb <span class="op">=</span> <span class="va">e_coli_regulondb</span>,
    dataset <span class="op">=</span> <span class="st">"GENE"</span>,
    attributes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"posleft"</span>, <span class="st">"posright"</span>, <span class="st">"strand"</span>, <span class="st">"name"</span>, <span class="st">"dna_sequence"</span><span class="op">)</span>,
    filters <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="st">"name"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"araC"</span>, <span class="st">"crp"</span>, <span class="st">"lacI"</span><span class="op">)</span><span class="op">)</span>
<span class="op">)</span>
<span class="va">res_dnastring</span> <span class="op">&lt;-</span>
    <span class="fu"><a href="../reference/convert_to_biostrings.html">convert_to_biostrings</a></span><span class="op">(</span><span class="va">res_dnastring</span>, seq_type <span class="op">=</span> <span class="st">"DNA"</span><span class="op">)</span>
<span class="va">res_dnastring</span>
<span class="co">#&gt; DNAStringSet object of length 3:</span>
<span class="co">#&gt;     width seq</span>
<span class="co">#&gt; [1]   879 ATGGCTGAAGCGCAAAATGATCCCCTGCTGCCGG...AAAGTGAATGATGTAGCCGTCAAGTTGTCATAA</span>
<span class="co">#&gt; [2]   633 ATGGTGCTTGGCAAACCGCAAACAGACCCGACTC...GGTAAAACCATCGTCGTTTACGGCACTCGTTAA</span>
<span class="co">#&gt; [3]  1083 GTGAAACCAGTAACGTTATACGATGTCGCAGAGT...CGACAGGTTTCCCGACTGGAAAGCGGGCAGTGA</span>
<span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu">mcols</span><span class="op">(</span><span class="va">res_dnastring</span><span class="op">)</span>
<span class="co">#&gt; regulondb_result with 3 rows and 4 columns</span>
<span class="co">#&gt;     posleft  posright      strand        name</span>
<span class="co">#&gt;   &lt;integer&gt; &lt;integer&gt; &lt;character&gt; &lt;character&gt;</span>
<span class="co">#&gt; 1     70387     71265     forward        araC</span>
<span class="co">#&gt; 2   3486120   3486752     forward         crp</span>
<span class="co">#&gt; 3    366428    367510     reverse        lacI</span></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit">
<span class="va">res_prodstring</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_dataset.html">get_dataset</a></span><span class="op">(</span>
    regulondb <span class="op">=</span> <span class="va">e_coli_regulondb</span>,
    dataset <span class="op">=</span> <span class="st">"GENE"</span>,
    attributes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"posleft"</span>, <span class="st">"posright"</span>, <span class="st">"strand"</span>, <span class="st">"name"</span>, <span class="st">"product_sequence"</span><span class="op">)</span>,
    filters <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="st">"name"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"araC"</span>, <span class="st">"crp"</span>, <span class="st">"lacI"</span><span class="op">)</span><span class="op">)</span>
<span class="op">)</span>
<span class="va">res_prodstring</span> <span class="op">&lt;-</span>
    <span class="fu"><a href="../reference/convert_to_biostrings.html">convert_to_biostrings</a></span><span class="op">(</span><span class="va">res_prodstring</span>, seq_type <span class="op">=</span> <span class="st">"product"</span><span class="op">)</span>
<span class="fu">mcols</span><span class="op">(</span><span class="va">res_prodstring</span><span class="op">)</span>
<span class="co">#&gt; regulondb_result with 3 rows and 4 columns</span>
<span class="co">#&gt;     posleft  posright      strand        name</span>
<span class="co">#&gt;   &lt;integer&gt; &lt;integer&gt; &lt;character&gt; &lt;character&gt;</span>
<span class="co">#&gt; 1     70387     71265     forward        araC</span>
<span class="co">#&gt; 2   3486120   3486752     forward         crp</span>
<span class="co">#&gt; 3    366428    367510     reverse        lacI</span></pre></div>
<p>As with the <strong>GRanges</strong> output mentioned above, it is possible for the output of <code><a href="../reference/get_dataset.html">get_dataset()</a></code> to be a <strong>DNAStringSet</strong> object by specifying the parameter <code>output_format="DNAStringSet"</code> or a <strong>BStringSet</strong> object by specifying <code>output_format="BStringSet"</code>. Note that the functions to convert <strong>regulondb_result</strong> objects will throw errors if there is insufficient information for the coercion to occur. For example, we will get an error if we try to convert into a <strong>GRanges</strong> object when genomic coordinates are missing from the <strong>regulondb_result</strong> object.</p>
</div>
<div id="building-your-own-queries" class="section level1">
<h1 class="hasAnchor">
<a href="#building-your-own-queries" class="anchor"></a>Building your own queries</h1>
<p>In the <em>regutools</em> package, we have implemented features that are commonly used when querying data from databases: filtering results by partial matching, filtering by numeric intervals, and building complex queries.</p>
<div id="partial-matching" class="section level2">
<h2 class="hasAnchor">
<a href="#partial-matching" class="anchor"></a>Partial matching</h2>
<p>The code below illustrates the concept of partial matching, in which by setting the parameter <code>partialmatch=</code> to <code>"name"</code>, the query returns all the gene name in which the word <em>ara</em> is contained.</p>
<div class="sourceCode" id="cb13"><pre class="downlit">
<span class="fu"><a href="../reference/get_dataset.html">get_dataset</a></span><span class="op">(</span>
    <span class="va">e_coli_regulondb</span>,
    attributes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"posright"</span>, <span class="st">"name"</span><span class="op">)</span>,
    filters <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="st">"name"</span> <span class="op">=</span> <span class="st">"ara"</span><span class="op">)</span>,
    partialmatch <span class="op">=</span> <span class="st">"name"</span>,
    dataset <span class="op">=</span> <span class="st">"GENE"</span>
<span class="op">)</span>
<span class="co">#&gt; regulondb_result with 13 rows and 2 columns</span>
<span class="co">#&gt;      posright        name</span>
<span class="co">#&gt;     &lt;integer&gt; &lt;character&gt;</span>
<span class="co">#&gt; 1       68337        araA</span>
<span class="co">#&gt; 2       70048        araB</span>
<span class="co">#&gt; 3       71265        araC</span>
<span class="co">#&gt; 4       66550        araD</span>
<span class="co">#&gt; 5     2982182        araE</span>
<span class="co">#&gt; ...       ...         ...</span>
<span class="co">#&gt; 9      412481        araJ</span>
<span class="co">#&gt; 10      30799        carA</span>
<span class="co">#&gt; 11    2917813        barA</span>
<span class="co">#&gt; 12    1619957        marA</span>
<span class="co">#&gt; 13     939337        rarA</span></pre></div>
<p>Note that setting the parameter <code>partialmatch=</code> to <code>NULL</code> will only return genes where the name string is identical to <em>ara</em>.</p>
</div>
<div id="filtering-by-numeric-intervals" class="section level2">
<h2 class="hasAnchor">
<a href="#filtering-by-numeric-intervals" class="anchor"></a>Filtering by numeric intervals</h2>
<p>In addition to partial matching, queries can be filtered by numeric intervals. For example, in the code below, the parameter <code>interv=</code> is set to <code>"posright"</code>. By doing this assignment, we are specifying that the values for <code>"posright"</code> must lie between the values of <code>posright</code> specified in the <code>filter=</code> parameter. Thus, the result of this query will be genes whose right positions lie between the coordinates 2000 and 4000000. Note that the use of the <code>interv=</code> parameter in the code below is equivalent to setting the parameter <code>output_format=</code> to <code>"GRanges"</code> and further subsetting the <em>GRanges</em> object using the function <code>subsetByOverlaps()</code>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit">
<span class="fu"><a href="../reference/get_dataset.html">get_dataset</a></span><span class="op">(</span>
    <span class="va">e_coli_regulondb</span>,
    attributes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"name"</span>, <span class="st">"strand"</span>, <span class="st">"posright"</span>, <span class="st">"product_name"</span><span class="op">)</span>,
    dataset <span class="op">=</span> <span class="st">"GENE"</span>,
    filters <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>posright <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"2000"</span>, <span class="st">"4000000"</span><span class="op">)</span><span class="op">)</span>,
    interval <span class="op">=</span> <span class="st">"posright"</span>
<span class="op">)</span>
<span class="co">#&gt; regulondb_result with 3963 rows and 4 columns</span>
<span class="co">#&gt;             name      strand  posright           product_name</span>
<span class="co">#&gt;      &lt;character&gt; &lt;character&gt; &lt;integer&gt;            &lt;character&gt;</span>
<span class="co">#&gt; 1           modB     forward    796551 molybdate ABC transp..</span>
<span class="co">#&gt; 2           cysZ     forward   2532224 sulfate:H&lt;sup&gt;+&lt;/sup..</span>
<span class="co">#&gt; 3            dfp     forward   3813951 fused 4'-phosphopant..</span>
<span class="co">#&gt; 4           hisM     reverse   2425233 lysine/arginine/orni..</span>
<span class="co">#&gt; 5           rhsE     forward   1529938 RhsE protein in rhs ..</span>
<span class="co">#&gt; ...          ...         ...       ...                    ...</span>
<span class="co">#&gt; 3959        yqfH     reverse   3033010           protein YqfH</span>
<span class="co">#&gt; 3960        yliM     forward    850397           protein YliM</span>
<span class="co">#&gt; 3961        ynfS     forward   1642211 Qin prophage; protei..</span>
<span class="co">#&gt; 3962        ylcJ     forward    568844           protein YlcJ</span>
<span class="co">#&gt; 3963        sdhX     forward    765150 small regulatory RNA..</span></pre></div>
</div>
<div id="retrieving-genomic-elements" class="section level2">
<h2 class="hasAnchor">
<a href="#retrieving-genomic-elements" class="anchor"></a>Retrieving genomic elements</h2>
<p>Based on genomic coordinates, the code below retrieves all genomic elements whose positions lie between the coordinates provided as a <em>GRanges</em> object. If no aditional parameters are provided, the result will retrieve genes that relies within the first 5000pb from the E. coli genome.</p>
<div class="sourceCode" id="cb15"><pre class="downlit">
<span class="fu"><a href="../reference/get_dna_objects.html">get_dna_objects</a></span><span class="op">(</span><span class="va">e_coli_regulondb</span><span class="op">)</span>
<span class="co">#&gt; GRanges object with 3 ranges and 4 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |           id        type        name</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |  &lt;character&gt; &lt;character&gt; &lt;character&gt;</span>
<span class="co">#&gt;   [1]   E.coli  337-2799      + | ECK120000987        gene        thrA</span>
<span class="co">#&gt;   [2]   E.coli 2801-3733      + | ECK120000988        gene        thrB</span>
<span class="co">#&gt;   [3]   E.coli   190-255      + | ECK120001251        gene        thrL</span>
<span class="co">#&gt;                  description</span>
<span class="co">#&gt;                  &lt;character&gt;</span>
<span class="co">#&gt;   [1] fused aspartate kina..</span>
<span class="co">#&gt;   [2]      homoserine kinase</span>
<span class="co">#&gt;   [3] &lt;i&gt;thr&lt;/i&gt; operon le..</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></pre></div>
<p>Especific genomic positions can be provided within the parameter <code>grange</code>. It is important to provide a genomic range that covers as minimal the length of one genomic element.</p>
<div class="sourceCode" id="cb16"><pre class="downlit">
<span class="va">grange</span> <span class="op">&lt;-</span> <span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span>
    <span class="st">"chr"</span>,
    <span class="fu">IRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html">IRanges</a></span><span class="op">(</span><span class="fl">5000</span>, <span class="fl">10000</span><span class="op">)</span>
<span class="op">)</span>
<span class="fu"><a href="../reference/get_dna_objects.html">get_dna_objects</a></span><span class="op">(</span><span class="va">e_coli_regulondb</span>, <span class="va">grange</span><span class="op">)</span>
<span class="co">#&gt; GRanges object with 3 ranges and 4 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |           id        type        name</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |  &lt;character&gt; &lt;character&gt; &lt;character&gt;</span>
<span class="co">#&gt;   [1]   E.coli  337-2799      + | ECK120000987        gene        thrA</span>
<span class="co">#&gt;   [2]   E.coli 2801-3733      + | ECK120000988        gene        thrB</span>
<span class="co">#&gt;   [3]   E.coli   190-255      + | ECK120001251        gene        thrL</span>
<span class="co">#&gt;                  description</span>
<span class="co">#&gt;                  &lt;character&gt;</span>
<span class="co">#&gt;   [1] fused aspartate kina..</span>
<span class="co">#&gt;   [2]      homoserine kinase</span>
<span class="co">#&gt;   [3] &lt;i&gt;thr&lt;/i&gt; operon le..</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></pre></div>
<p>Aditional genomic elements such as “-35 promoter box”, “gene”, “promoter”, “Regulatory Interaction”, “sRNA interaction”, or “terminator” can be selected.</p>
<div class="sourceCode" id="cb17"><pre class="downlit">
<span class="va">grange</span> <span class="op">&lt;-</span> <span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span>
    <span class="st">"chr"</span>,
    <span class="fu">IRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html">IRanges</a></span><span class="op">(</span><span class="fl">5000</span>, <span class="fl">10000</span><span class="op">)</span>
<span class="op">)</span>
<span class="fu"><a href="../reference/get_dna_objects.html">get_dna_objects</a></span><span class="op">(</span><span class="va">e_coli_regulondb</span>, <span class="va">grange</span>, elements <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"gene"</span>, <span class="st">"promoter"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; GRanges object with 19 ranges and 4 metadata columns:</span>
<span class="co">#&gt;        seqnames    ranges strand |           id        type</span>
<span class="co">#&gt;           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |  &lt;character&gt; &lt;character&gt;</span>
<span class="co">#&gt;    [1]   E.coli  337-2799      + | ECK120000987        gene</span>
<span class="co">#&gt;    [2]   E.coli 2801-3733      + | ECK120000988        gene</span>
<span class="co">#&gt;    [3]   E.coli   190-255      + | ECK120001251        gene</span>
<span class="co">#&gt;    [4]   E.coli       148      + | ECK120010236    promoter</span>
<span class="co">#&gt;    [5]   E.coli        38      + | ECK125230824    promoter</span>
<span class="co">#&gt;    ...      ...       ...    ... .          ...         ...</span>
<span class="co">#&gt;   [15]   E.coli      3066      + | ECK125230834    promoter</span>
<span class="co">#&gt;   [16]   E.coli      3368      + | ECK125230835    promoter</span>
<span class="co">#&gt;   [17]   E.coli      3396      + | ECK125230836    promoter</span>
<span class="co">#&gt;   [18]   E.coli      3401      + | ECK125230837    promoter</span>
<span class="co">#&gt;   [19]   E.coli      4241      + | ECK125230838    promoter</span>
<span class="co">#&gt;                          name            description</span>
<span class="co">#&gt;                   &lt;character&gt;            &lt;character&gt;</span>
<span class="co">#&gt;    [1]                   thrA fused aspartate kina..</span>
<span class="co">#&gt;    [2]                   thrB      homoserine kinase</span>
<span class="co">#&gt;    [3]                   thrL &lt;i&gt;thr&lt;/i&gt; operon le..</span>
<span class="co">#&gt;    [4] thrLp promoter with .. thrLp promoter with ..</span>
<span class="co">#&gt;    [5] TSS_1 promoter with .. TSS_1 promoter with ..</span>
<span class="co">#&gt;    ...                    ...                    ...</span>
<span class="co">#&gt;   [15] TSS_12 promoter with.. TSS_12 promoter with..</span>
<span class="co">#&gt;   [16] TSS_13 promoter with.. TSS_13 promoter with..</span>
<span class="co">#&gt;   [17] TSS_14 promoter with.. TSS_14 promoter with..</span>
<span class="co">#&gt;   [18] TSS_15 promoter with.. TSS_15 promoter with..</span>
<span class="co">#&gt;   [19] TSS_16 promoter with.. TSS_16 promoter with..</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></pre></div>
<p>Evenmore, the genomic elements retrieved can be observed in a Genome Browser-like plot. The genomic elements are annotated using a UCSC genome as reference, it is important to provide a valid chromosome name for annotation purpose.</p>
<div class="sourceCode" id="cb18"><pre class="downlit">
<span class="va">e_coli_regulondb</span> <span class="op">&lt;-</span>
    <span class="fu"><a href="../reference/regulondb.html">regulondb</a></span><span class="op">(</span>
        database_conn <span class="op">=</span> <span class="va">regulondb_conn</span>,
        organism <span class="op">=</span> <span class="st">"chr"</span>,
        database_version <span class="op">=</span> <span class="st">"1"</span>,
        genome_version <span class="op">=</span> <span class="st">"1"</span>
    <span class="op">)</span>

<span class="va">grange</span> <span class="op">&lt;-</span> <span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span><span class="st">"chr"</span>, <span class="fu">IRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html">IRanges</a></span><span class="op">(</span><span class="fl">5000</span>, <span class="fl">10000</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="../reference/plot_dna_objects.html">plot_dna_objects</a></span><span class="op">(</span><span class="va">e_coli_regulondb</span>, <span class="va">grange</span>, elements <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"gene"</span>, <span class="st">"promoter"</span><span class="op">)</span><span class="op">)</span></pre></div>
<p><img src="regutools_files/figure-html/plot_elements-1.png" width="700"></p>
</div>
<div id="complex-filters" class="section level2">
<h2 class="hasAnchor">
<a href="#complex-filters" class="anchor"></a>Complex filters</h2>
<p>The examples so far have considered queries in which the results are filtered according to single fields from tables. In order to build queries with filters from more than one field, several filters names can be passed as a list to the <code>filters=</code> argument. Additionally the <code>and=</code> argument is used to specify whether the filtering conditions of the result of the query must be satisfied (<code>and=TRUE</code>) or if satisfying a single condition is enough (<code>and=FALSE</code>).</p>
<p>For example, the code below extracts the genes where either <code>name</code> or <code>product_name</code> contain the word <em>Ara</em> or <em>Ara</em>, respectively, if the gene is in the forward strand or if the right position of the gene is between 2000 and 40000000.</p>
<div class="sourceCode" id="cb19"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span>
    <span class="fu"><a href="../reference/get_dataset.html">get_dataset</a></span><span class="op">(</span>
        <span class="va">e_coli_regulondb</span>,
        attributes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"name"</span>, <span class="st">"strand"</span>, <span class="st">"posright"</span>, <span class="st">"product_name"</span><span class="op">)</span>,
        dataset <span class="op">=</span> <span class="st">"GENE"</span>,
        filters <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
            name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ARA"</span><span class="op">)</span>,
            product_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Ara"</span><span class="op">)</span>,
            strand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"forward"</span><span class="op">)</span>,
            posright <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"2000"</span>, <span class="st">"4000000"</span><span class="op">)</span>
        <span class="op">)</span>,
        and <span class="op">=</span> <span class="cn">FALSE</span>,
        partialmatch <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"name"</span>, <span class="st">"product_name"</span><span class="op">)</span>,
        interval <span class="op">=</span> <span class="st">"posright"</span>
    <span class="op">)</span>
<span class="op">)</span>
<span class="co">#&gt; [1] 2331</span></pre></div>
<p>The query below, which is identical to the query above except the <code>and=</code> is set to <code>TRUE</code>, returns the genes where all of the specified conditions are satisfied.</p>
<div class="sourceCode" id="cb20"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span>
    <span class="fu"><a href="../reference/get_dataset.html">get_dataset</a></span><span class="op">(</span>
        <span class="va">e_coli_regulondb</span>,
        attributes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"name"</span>, <span class="st">"strand"</span>, <span class="st">"posright"</span>, <span class="st">"product_name"</span><span class="op">)</span>,
        dataset <span class="op">=</span> <span class="st">"GENE"</span>,
        filters <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
            name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ARA"</span><span class="op">)</span>,
            product_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Ara"</span><span class="op">)</span>,
            strand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"forward"</span><span class="op">)</span>,
            posright <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"2000"</span>, <span class="st">"4000000"</span><span class="op">)</span>
        <span class="op">)</span>,
        and <span class="op">=</span> <span class="cn">TRUE</span>,
        partialmatch <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"name"</span>, <span class="st">"product_name"</span><span class="op">)</span>,
        interval <span class="op">=</span> <span class="st">"posright"</span>
    <span class="op">)</span>
<span class="op">)</span>
<span class="co">#&gt; [1] 3</span></pre></div>
</div>
</div>
<div id="functions-with-implement-popular-queries" class="section level1">
<h1 class="hasAnchor">
<a href="#functions-with-implement-popular-queries" class="anchor"></a>Functions with implement popular queries</h1>
<p>The regutools package provides functions that encode the most frequent queries that are submitted to the <em>RegulonDB</em> web resource.</p>
<div id="extracting-regulatory-networks" class="section level2">
<h2 class="hasAnchor">
<a href="#extracting-regulatory-networks" class="anchor"></a>Extracting regulatory networks</h2>
<p>One of the most important information that the <em>RegulonDB</em> database contains are manually curated regulatory networks. The <em>regutools</em> package exports the function <code><a href="../reference/get_gene_regulators.html">get_gene_regulators()</a></code>, which inputs a vector of gene names and outputs information about what are the transcription factors that regulate such genes together with the regulatory effect (activator, repressor or dual).</p>
<div class="sourceCode" id="cb21"><pre class="downlit">
<span class="fu"><a href="../reference/get_gene_regulators.html">get_gene_regulators</a></span><span class="op">(</span><span class="va">e_coli_regulondb</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"araC"</span>, <span class="st">"fis"</span>, <span class="st">"crp"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; regulondb_result with 9 rows and 3 columns</span>
<span class="co">#&gt;         genes  regulators      effect</span>
<span class="co">#&gt;   &lt;character&gt; &lt;character&gt; &lt;character&gt;</span>
<span class="co">#&gt; 1         crp         Fis           -</span>
<span class="co">#&gt; 2         fis         Fis           -</span>
<span class="co">#&gt; 3        araC         CRP           +</span>
<span class="co">#&gt; 4         crp         CRP         +/-</span>
<span class="co">#&gt; 5         fis         CRP         +/-</span>
<span class="co">#&gt; 6        araC        AraC         +/-</span>
<span class="co">#&gt; 7         crp         Cra           +</span>
<span class="co">#&gt; 8        araC        XylR           -</span>
<span class="co">#&gt; 9         fis         IHF           +</span></pre></div>
<p>Similarly, the <code><a href="../reference/get_regulatory_network.html">get_regulatory_network()</a></code> function retrieves all the regulatory network from a <strong>regulondb</strong> object. By default, the output will be a list of transcription factor-gene pairs, indicating which transcription factor regulates which gene.</p>
<div class="sourceCode" id="cb22"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/get_regulatory_network.html">get_regulatory_network</a></span><span class="op">(</span><span class="va">e_coli_regulondb</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; regulondb_result with 6 rows and 3 columns</span>
<span class="co">#&gt;     regulator   regulated      effect</span>
<span class="co">#&gt;   &lt;character&gt; &lt;character&gt; &lt;character&gt;</span>
<span class="co">#&gt; 1         Fis        adhE           +</span>
<span class="co">#&gt; 2         Fis        osmE           -</span>
<span class="co">#&gt; 3         Fis        ansB           -</span>
<span class="co">#&gt; 4         Fis        apaG           +</span>
<span class="co">#&gt; 5         Fis        apaH           +</span>
<span class="co">#&gt; 6         Fis        bglB           -</span></pre></div>
<p>But users can also set the parameter <code>type=</code> to <code>"GENE-GENE"</code> or <code>"TF-GENE"</code> to retrieve gene-gene regulatory networks or transcription factor-transcription factor regulatory networks, respectively.</p>
<p>Users can also use the <code><a href="../reference/get_regulatory_summary.html">get_regulatory_summary()</a></code> to retrieve a summary of the transcription factor regulated the expression of a set of given genes. The parameter <code>gene_regulators</code> can receive either a vector of gene names or the output of a call to the function <code><a href="../reference/get_gene_regulators.html">get_gene_regulators()</a></code>. The resulting output is a <em>regulondb_result</em> object in which each row shows a transcription factor, and the columns display information about the number, percentage and regulatory activity exerted to the genes.</p>
<div class="sourceCode" id="cb23"><pre class="downlit">
<span class="fu"><a href="../reference/get_regulatory_summary.html">get_regulatory_summary</a></span><span class="op">(</span><span class="va">e_coli_regulondb</span>,
    gene_regulators <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"araC"</span>, <span class="st">"modB"</span><span class="op">)</span>
<span class="op">)</span>
<span class="co">#&gt; regulondb_result with 4 rows and 7 columns</span>
<span class="co">#&gt;            TF Regulated_genes_per_TF     Percent   Activator   Repressor</span>
<span class="co">#&gt;   &lt;character&gt;            &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;</span>
<span class="co">#&gt; 1        AraC                      1          20           0           0</span>
<span class="co">#&gt; 2         CRP                      2          40           2           0</span>
<span class="co">#&gt; 3        ModE                      1          20           0           1</span>
<span class="co">#&gt; 4        XylR                      1          20           0           1</span>
<span class="co">#&gt;          Dual Regulated_genes</span>
<span class="co">#&gt;   &lt;character&gt;     &lt;character&gt;</span>
<span class="co">#&gt; 1           1            araC</span>
<span class="co">#&gt; 2           0      modB, araC</span>
<span class="co">#&gt; 3           0            modB</span>
<span class="co">#&gt; 4           0            araC</span></pre></div>
</div>
<div id="visualizing-networks-using-cytoscape" class="section level2">
<h2 class="hasAnchor">
<a href="#visualizing-networks-using-cytoscape" class="anchor"></a>Visualizing networks using cytoscape</h2>
<p>Software tools such as <a href="https://cytoscape.org/index.html">Cytoscape</a> are often useful for interactive exploration of data and visualization of networks. The function <code><a href="../reference/get_regulatory_network.html">get_regulatory_network()</a></code> has a parameter <code>cytograph=</code> that if set to <code>TRUE</code>, it will visualize the network in a cytoscape session. Of note, this feature will only work if the user has Cytoscape open in their computer.</p>
<div class="sourceCode" id="cb24"><pre class="downlit">
<span class="fu"><a href="../reference/get_regulatory_network.html">get_regulatory_network</a></span><span class="op">(</span><span class="va">e_coli_regulondb</span>, cytograph <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></pre></div>
</div>
<div id="transcription-factor-binding-sites" class="section level2">
<h2 class="hasAnchor">
<a href="#transcription-factor-binding-sites" class="anchor"></a>Transcription factor binding sites</h2>
<p>Another common request to the <em>RegulonDB</em> database is to obtain the genomic coordinates and sequences of the DNA binding sites for a given transcription factor. This query is implemented in the function <code><a href="../reference/get_binding_sites.html">get_binding_sites()</a></code>, in which the results are formatted according to the parameter <code>output_format=</code> as either a <em>GRanges</em> object or a <em>Biostrings</em> object.</p>
<div class="sourceCode" id="cb25"><pre class="downlit">
<span class="fu"><a href="../reference/get_binding_sites.html">get_binding_sites</a></span><span class="op">(</span><span class="va">e_coli_regulondb</span>, transcription_factor <span class="op">=</span> <span class="st">"AraC"</span><span class="op">)</span>
<span class="co">#&gt; GRanges object with 15 ranges and 1 metadata column:</span>
<span class="co">#&gt;                          seqnames          ranges strand |</span>
<span class="co">#&gt;                             &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |</span>
<span class="co">#&gt;   ECK120015742-araB-araC      chr     70110-70126      + |</span>
<span class="co">#&gt;   ECK120012328-araB-araC      chr     70131-70147      + |</span>
<span class="co">#&gt;   ECK120012320-araB-araC      chr     70184-70200      - |</span>
<span class="co">#&gt;   ECK120012323-araB-araC      chr     70205-70221      - |</span>
<span class="co">#&gt;   ECK120012603-araB-araC      chr     70342-70358      - |</span>
<span class="co">#&gt;                      ...      ...             ...    ... .</span>
<span class="co">#&gt;        ECK120012333-araF      chr 1986396-1986412      - |</span>
<span class="co">#&gt;        ECK120012915-araE      chr 2982244-2982260      - |</span>
<span class="co">#&gt;        ECK120012913-araE      chr 2982265-2982281      - |</span>
<span class="co">#&gt;        ECK125108641-xylA      chr 3730824-3730840      - |</span>
<span class="co">#&gt;        ECK125108643-xylA      chr 3730847-3730863      - |</span>
<span class="co">#&gt;                                        sequence</span>
<span class="co">#&gt;                                     &lt;character&gt;</span>
<span class="co">#&gt;   ECK120015742-araB-araC ataaaaagcgTCAGGTAGGA..</span>
<span class="co">#&gt;   ECK120012328-araB-araC ccgctaatctTATGGATAAA..</span>
<span class="co">#&gt;   ECK120012320-araB-araC tctataatcaCGGCAGAAAA..</span>
<span class="co">#&gt;   ECK120012323-araB-araC caaaaacgcgTAACAAAAGT..</span>
<span class="co">#&gt;   ECK120012603-araB-araC attcagagaaGAAACCAATT..</span>
<span class="co">#&gt;                      ...                    ...</span>
<span class="co">#&gt;        ECK120012333-araF ccaaagacaaCAAGGATTTC..</span>
<span class="co">#&gt;        ECK120012915-araE tccatatttaTGCTGTTTCC..</span>
<span class="co">#&gt;        ECK120012913-araE cgacatgtcgCAGCAATTTA..</span>
<span class="co">#&gt;        ECK125108641-xylA taacataattGAGCAACTGA..</span>
<span class="co">#&gt;        ECK125108643-xylA attatctcaaTAGCAGTGTG..</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="fu"><a href="../reference/get_binding_sites.html">get_binding_sites</a></span><span class="op">(</span><span class="va">e_coli_regulondb</span>,
    transcription_factor <span class="op">=</span> <span class="st">"AraC"</span>,
    output_format <span class="op">=</span> <span class="st">"Biostrings"</span>
<span class="op">)</span>
<span class="co">#&gt; DNAStringSet object of length 15:</span>
<span class="co">#&gt;      width seq                                              names               </span>
<span class="co">#&gt;  [1]    37 ATAAAAAGCGTCAGGTAGGATCCGCTAATCTTATGGA            ECK120015742-araB...</span>
<span class="co">#&gt;  [2]    37 CCGCTAATCTTATGGATAAAAATGCTATGGCATAGCA            ECK120012328-araB...</span>
<span class="co">#&gt;  [3]    37 TCTATAATCACGGCAGAAAAGTCCACATTGATTATTT            ECK120012320-araB...</span>
<span class="co">#&gt;  [4]    37 CAAAAACGCGTAACAAAAGTGTCTATAATCACGGCAG            ECK120012323-araB...</span>
<span class="co">#&gt;  [5]    37 ATTCAGAGAAGAAACCAATTGTCCATATTGCATCAGA            ECK120012603-araB...</span>
<span class="co">#&gt;  ...   ... ...</span>
<span class="co">#&gt; [11]    37 CCAAAGACAACAAGGATTTCCAGGCTAATCTTATGGA            ECK120012333-araF</span>
<span class="co">#&gt; [12]    37 TCCATATTTATGCTGTTTCCGACCTGACACCTGCGTG            ECK120012915-araE</span>
<span class="co">#&gt; [13]    37 CGACATGTCGCAGCAATTTAATCCATATTTATGCTGT            ECK120012913-araE</span>
<span class="co">#&gt; [14]    37 TAACATAATTGAGCAACTGAAAGGGAGTGCCCAATAT            ECK125108641-xylA</span>
<span class="co">#&gt; [15]    37 ATTATCTCAATAGCAGTGTGAAATAACATAATTGAGC            ECK125108643-xylA</span></pre></div>
</div>
</div>
<div id="a-note-about-cdsb" class="section level1">
<h1 class="hasAnchor">
<a href="#a-note-about-cdsb" class="anchor"></a>A note about CDSB</h1>
<p>This was a project accomplished by members of the <a href="https://comunidadbioinfo.github.io/">Community of Bioinformatics Software Developers</a> (CDSB in Spanish). In part CDSB was formed to help R users in Latin America become R/<a href="http://bioconductor.org/">Bioconductor</a> developers. For more information about CDSB, the CDSB workshops or its online community, please check the <a href="https://comunidadbioinfo.github.io/">CDSB website</a> which is available in both Spanish and English.</p>
<p><a href="https://comunidadbioinfo.github.io/"><img src="https://comunidadbioinfo.github.io/img/Logo_texto-768x107.png"></a></p>
</div>
<div id="reproducibility" class="section level1">
<h1 class="hasAnchor">
<a href="#reproducibility" class="anchor"></a>Reproducibility</h1>
<p>The <em><a href="https://bioconductor.org/packages/3.12/regutools">regutools</a></em> package <a id="cite-Chavez_2020a"></a>(<a href="http://www.bioconductor.org/packages/regutools">Chávez, Barberena-Jonas, Sotelo-Fonseca, Alquicira-Hernandez, et al., 2020</a>) was made possible thanks to:</p>
<ul>
<li>R <a id="cite-R_Core_Team_2020"></a>(<a href="https://www.R-project.org/">R Core Team, 2020</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/AnnotationDbi">AnnotationDbi</a></em> <a id="cite-Pages_2020"></a>(<a href="#bib-Pages_2020">Pagès, Carlson, Falcon, and Li, 2020</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/AnnotationHub">AnnotationHub</a></em> (<a href="#bib-Morgan_2020">Morgan and Shepherd, 2020</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/BiocFileCache">BiocFileCache</a></em> <a id="cite-Shepherd_2020"></a>(<a href="#bib-Shepherd_2020">Shepherd and Morgan, 2020</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/BiocStyle">BiocStyle</a></em> <a id="cite-Oles_2020"></a>(<a href="https://github.com/Bioconductor/BiocStyle">Oleś, Morgan, and Huber, 2020</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/Biostrings">Biostrings</a></em> <a id="cite-Pages_2020a"></a>(<a href="#bib-Pages_2020a">Pagès, Aboyoun, Gentleman, and DebRoy, 2020</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=DBI">DBI</a></em> <a id="cite-R_Special_Interest_Group_on_Databases_R-SIG-DB_2019"></a>(<a href="#bib-R_Special_Interest_Group_on_Databases_R-SIG-DB_2019">R Special Interest Group on Databases (R-SIG-DB), Wickham, and Müller, 2019</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/GenomicRanges">GenomicRanges</a></em> <a id="cite-Lawrence_2013"></a>(<a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118">Lawrence, Huber, Pagès, Aboyoun, et al., 2013</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/Gviz">Gviz</a></em> <a id="cite-Hahne_2016"></a>(<a href="http://dx.doi.org/10.1007/978-1-4939-3578-9_16">Hahne and Ivanek, 2016</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/IRanges">IRanges</a></em> <a id="cite-Lawrence_2013a"></a>(<a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118">Lawrence, Huber, Pagès, Aboyoun, et al., 2013</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitcitations">knitcitations</a></em> <a id="cite-Boettiger_2019"></a>(<a href="https://github.com/cboettig/knitcitations">Boettiger, 2019</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr">knitr</a></em> <a id="cite-Xie_2014"></a>(<a href="http://www.crcpress.com/product/isbn/9781466561595">Xie, 2014</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/RCy3">RCy3</a></em> <a id="cite-Gustavsen_2019"></a>(<a href="https://doi.org/10.12688/f1000research.20887.3">Gustavsen, A., Pai, Shraddha, et al., 2019</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown">rmarkdown</a></em> <a id="cite-Allaire_2020"></a>(<a href="https://github.com/rstudio/rmarkdown">Allaire, Xie, McPherson, Luraschi, et al., 2020</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RSQLite">RSQLite</a></em> <a id="cite-Muller_2020"></a>(<a href="#bib-Muller_2020">Müller, Wickham, James, and Falcon, 2020</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/S4Vectors">S4Vectors</a></em> <a id="cite-Pages_2020ab"></a>(<a href="#bib-Pages_2020ab">Pagès, Lawrence, and Aboyoun, 2020</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo">sessioninfo</a></em> <a id="cite-Csardi_2018"></a>(<a href="https://github.com/r-lib/sessioninfo#readme">Csárdi, core, Wickham, Chang, et al., 2018</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat">testthat</a></em> <a id="cite-Wickham_2011"></a>(<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf">Wickham, 2011</a>)</li>
</ul>
<p>Code for creating the vignette</p>
<div class="sourceCode" id="cb26"><pre class="downlit">
<span class="co">## Create the vignette</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/rstudio/rmarkdown">"rmarkdown"</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/rmarkdown/man/render.html">render</a></span><span class="op">(</span><span class="st">"regutools.Rmd"</span><span class="op">)</span><span class="op">)</span>

<span class="co">## Extract the R code</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://yihui.org/knitr/">"knitr"</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/knitr/man/knit.html">knit</a></span><span class="op">(</span><span class="st">"regutools.Rmd"</span>, tangle <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></pre></div>
<p>Date the vignette was generated.</p>
<pre><code>#&gt; [1] "2020-10-11 19:42:18 UTC"</code></pre>
<p>Wallclock time spent generating the vignette.</p>
<pre><code>#&gt; Time difference of 33.596 secs</code></pre>
<p><code>R</code> session information.</p>
<pre><code>#&gt; ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────
#&gt;  setting  value                       
#&gt;  version  R version 4.0.2 (2020-06-22)
#&gt;  os       Ubuntu 20.04.1 LTS          
#&gt;  system   x86_64, linux-gnu           
#&gt;  ui       X11                         
#&gt;  language (EN)                        
#&gt;  collate  en_US.UTF-8                 
#&gt;  ctype    en_US.UTF-8                 
#&gt;  tz       Etc/UTC                     
#&gt;  date     2020-10-11                  
#&gt; 
#&gt; ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────
#&gt;  package                * version    date       lib source                        
#&gt;  AnnotationDbi            1.51.3     2020-07-25 [1] Bioconductor                  
#&gt;  AnnotationFilter         1.13.0     2020-04-27 [1] Bioconductor                  
#&gt;  AnnotationHub            2.21.5     2020-08-26 [1] Bioconductor                  
#&gt;  askpass                  1.1        2019-01-13 [2] RSPM (R 4.0.0)                
#&gt;  assertthat               0.2.1      2019-03-21 [2] RSPM (R 4.0.0)                
#&gt;  backports                1.1.10     2020-09-15 [2] RSPM (R 4.0.2)                
#&gt;  base64enc                0.1-3      2015-07-28 [2] RSPM (R 4.0.0)                
#&gt;  bibtex                   0.4.2.3    2020-09-19 [1] RSPM (R 4.0.2)                
#&gt;  Biobase                  2.49.1     2020-09-03 [1] Bioconductor                  
#&gt;  BiocFileCache            1.13.1     2020-08-04 [1] Bioconductor                  
#&gt;  BiocGenerics           * 0.35.4     2020-06-04 [1] Bioconductor                  
#&gt;  BiocManager              1.30.10    2019-11-16 [2] CRAN (R 4.0.2)                
#&gt;  BiocParallel             1.23.2     2020-07-06 [1] Bioconductor                  
#&gt;  BiocStyle              * 2.17.1     2020-09-24 [1] Bioconductor                  
#&gt;  BiocVersion              3.12.0     2020-04-27 [2] Bioconductor                  
#&gt;  biomaRt                  2.45.5     2020-10-09 [1] Bioconductor                  
#&gt;  Biostrings             * 2.57.2     2020-06-09 [1] Bioconductor                  
#&gt;  biovizBase               1.37.0     2020-04-27 [1] Bioconductor                  
#&gt;  bit                      4.0.4      2020-08-04 [1] RSPM (R 4.0.2)                
#&gt;  bit64                    4.0.5      2020-08-30 [1] RSPM (R 4.0.2)                
#&gt;  bitops                   1.0-6      2013-08-17 [1] RSPM (R 4.0.0)                
#&gt;  blob                     1.2.1      2020-01-20 [1] RSPM (R 4.0.0)                
#&gt;  bookdown                 0.20       2020-06-23 [1] RSPM (R 4.0.2)                
#&gt;  BSgenome                 1.57.7     2020-10-07 [1] Bioconductor                  
#&gt;  checkmate                2.0.0      2020-02-06 [1] RSPM (R 4.0.0)                
#&gt;  cli                      2.0.2      2020-02-28 [2] RSPM (R 4.0.0)                
#&gt;  cluster                  2.1.0      2019-06-19 [3] CRAN (R 4.0.2)                
#&gt;  colorspace               1.4-1      2019-03-18 [1] RSPM (R 4.0.0)                
#&gt;  crayon                   1.3.4      2017-09-16 [2] RSPM (R 4.0.0)                
#&gt;  curl                     4.3        2019-12-02 [2] RSPM (R 4.0.0)                
#&gt;  data.table               1.13.0     2020-07-24 [1] RSPM (R 4.0.2)                
#&gt;  DBI                      1.1.0      2019-12-15 [1] RSPM (R 4.0.0)                
#&gt;  dbplyr                   1.4.4      2020-05-27 [1] RSPM (R 4.0.0)                
#&gt;  DelayedArray             0.15.16    2020-10-06 [1] Bioconductor                  
#&gt;  desc                     1.2.0      2018-05-01 [2] RSPM (R 4.0.0)                
#&gt;  dichromat                2.0-0      2013-01-24 [1] RSPM (R 4.0.0)                
#&gt;  digest                   0.6.25     2020-02-23 [2] RSPM (R 4.0.0)                
#&gt;  dplyr                    1.0.2      2020-08-18 [1] RSPM (R 4.0.2)                
#&gt;  ellipsis                 0.3.1      2020-05-15 [2] RSPM (R 4.0.0)                
#&gt;  ensembldb                2.13.1     2020-05-06 [1] Bioconductor                  
#&gt;  evaluate                 0.14       2019-05-28 [2] RSPM (R 4.0.0)                
#&gt;  fansi                    0.4.1      2020-01-08 [2] RSPM (R 4.0.0)                
#&gt;  fastmap                  1.0.1      2019-10-08 [1] RSPM (R 4.0.0)                
#&gt;  foreign                  0.8-80     2020-05-24 [3] CRAN (R 4.0.2)                
#&gt;  Formula                  1.2-3      2018-05-03 [1] RSPM (R 4.0.0)                
#&gt;  fs                       1.5.0      2020-07-31 [2] RSPM (R 4.0.2)                
#&gt;  generics                 0.0.2      2018-11-29 [1] RSPM (R 4.0.0)                
#&gt;  GenomeInfoDb             1.25.11    2020-09-03 [1] Bioconductor                  
#&gt;  GenomeInfoDbData         1.2.4      2020-10-10 [1] Bioconductor                  
#&gt;  GenomicAlignments        1.25.3     2020-06-09 [1] Bioconductor                  
#&gt;  GenomicFeatures          1.41.3     2020-09-03 [1] Bioconductor                  
#&gt;  GenomicRanges            1.41.6     2020-08-12 [1] Bioconductor                  
#&gt;  ggplot2                  3.3.2      2020-06-19 [1] RSPM (R 4.0.1)                
#&gt;  glue                     1.4.2      2020-08-27 [2] RSPM (R 4.0.2)                
#&gt;  graph                    1.67.1     2020-05-27 [1] Bioconductor                  
#&gt;  gridExtra                2.3        2017-09-09 [1] RSPM (R 4.0.0)                
#&gt;  gtable                   0.3.0      2019-03-25 [1] RSPM (R 4.0.0)                
#&gt;  Gviz                     1.33.2     2020-07-06 [1] Bioconductor                  
#&gt;  Hmisc                    4.4-1      2020-08-10 [1] RSPM (R 4.0.2)                
#&gt;  hms                      0.5.3      2020-01-08 [1] RSPM (R 4.0.0)                
#&gt;  htmlTable                2.1.0      2020-09-16 [1] RSPM (R 4.0.2)                
#&gt;  htmltools                0.5.0      2020-06-16 [2] RSPM (R 4.0.1)                
#&gt;  htmlwidgets              1.5.2      2020-10-03 [2] RSPM (R 4.0.2)                
#&gt;  httpuv                   1.5.4      2020-06-06 [1] RSPM (R 4.0.2)                
#&gt;  httr                     1.4.2      2020-07-20 [2] RSPM (R 4.0.2)                
#&gt;  igraph                   1.2.6      2020-10-06 [1] RSPM (R 4.0.2)                
#&gt;  interactiveDisplayBase   1.27.5     2020-06-01 [1] Bioconductor                  
#&gt;  IRanges                * 2.23.10    2020-06-13 [1] Bioconductor                  
#&gt;  jpeg                     0.1-8.1    2019-10-24 [1] RSPM (R 4.0.0)                
#&gt;  jsonlite                 1.7.1      2020-09-07 [2] RSPM (R 4.0.2)                
#&gt;  knitcitations          * 1.0.10     2019-09-15 [1] RSPM (R 4.0.0)                
#&gt;  knitr                    1.30       2020-09-22 [2] RSPM (R 4.0.2)                
#&gt;  later                    1.1.0.1    2020-06-05 [2] RSPM (R 4.0.2)                
#&gt;  lattice                  0.20-41    2020-04-02 [3] CRAN (R 4.0.2)                
#&gt;  latticeExtra             0.6-29     2019-12-19 [1] RSPM (R 4.0.0)                
#&gt;  lazyeval                 0.2.2      2019-03-15 [2] RSPM (R 4.0.0)                
#&gt;  lifecycle                0.2.0      2020-03-06 [2] RSPM (R 4.0.0)                
#&gt;  lubridate                1.7.9      2020-06-08 [1] RSPM (R 4.0.2)                
#&gt;  magrittr                 1.5        2014-11-22 [2] RSPM (R 4.0.0)                
#&gt;  Matrix                   1.2-18     2019-11-27 [3] CRAN (R 4.0.2)                
#&gt;  MatrixGenerics           1.1.4      2020-10-08 [1] Bioconductor                  
#&gt;  matrixStats              0.57.0     2020-09-25 [1] RSPM (R 4.0.2)                
#&gt;  memoise                  1.1.0      2017-04-21 [2] RSPM (R 4.0.0)                
#&gt;  mime                     0.9        2020-02-04 [2] RSPM (R 4.0.0)                
#&gt;  munsell                  0.5.0      2018-06-12 [1] RSPM (R 4.0.0)                
#&gt;  nnet                     7.3-14     2020-04-26 [3] CRAN (R 4.0.2)                
#&gt;  openssl                  1.4.3      2020-09-18 [2] RSPM (R 4.0.2)                
#&gt;  pillar                   1.4.6      2020-07-10 [2] RSPM (R 4.0.2)                
#&gt;  pkgconfig                2.0.3      2019-09-22 [2] RSPM (R 4.0.0)                
#&gt;  pkgdown                  1.6.1.9000 2020-10-10 [1] Github (r-lib/pkgdown@e1e46f2)
#&gt;  plyr                     1.8.6      2020-03-03 [1] RSPM (R 4.0.2)                
#&gt;  png                      0.1-7      2013-12-03 [1] RSPM (R 4.0.0)                
#&gt;  prettyunits              1.1.1      2020-01-24 [2] RSPM (R 4.0.0)                
#&gt;  progress                 1.2.2      2019-05-16 [1] RSPM (R 4.0.0)                
#&gt;  promises                 1.1.1      2020-06-09 [2] RSPM (R 4.0.2)                
#&gt;  ProtGenerics             1.21.0     2020-04-27 [1] Bioconductor                  
#&gt;  purrr                    0.3.4      2020-04-17 [2] RSPM (R 4.0.0)                
#&gt;  R.methodsS3              1.8.1      2020-08-26 [1] RSPM (R 4.0.2)                
#&gt;  R.oo                     1.24.0     2020-08-26 [1] RSPM (R 4.0.2)                
#&gt;  R.utils                  2.10.1     2020-08-26 [1] RSPM (R 4.0.2)                
#&gt;  R6                       2.4.1      2019-11-12 [2] RSPM (R 4.0.0)                
#&gt;  ragg                     0.4.0      2020-10-05 [1] RSPM (R 4.0.2)                
#&gt;  rappdirs                 0.3.1      2016-03-28 [1] RSPM (R 4.0.0)                
#&gt;  RColorBrewer             1.1-2      2014-12-07 [1] RSPM (R 4.0.0)                
#&gt;  Rcpp                     1.0.5      2020-07-06 [2] RSPM (R 4.0.2)                
#&gt;  RCurl                    1.98-1.2   2020-04-18 [1] RSPM (R 4.0.0)                
#&gt;  RCy3                     2.9.2      2020-08-21 [1] Bioconductor                  
#&gt;  RefManageR               1.2.12     2019-04-03 [1] RSPM (R 4.0.0)                
#&gt;  regutools              * 1.1.3      2020-10-11 [1] Bioconductor                  
#&gt;  RJSONIO                  1.3-1.4    2020-01-15 [1] RSPM (R 4.0.0)                
#&gt;  rlang                    0.4.8      2020-10-08 [2] RSPM (R 4.0.2)                
#&gt;  rmarkdown                2.4        2020-09-30 [1] RSPM (R 4.0.2)                
#&gt;  rpart                    4.1-15     2019-04-12 [3] CRAN (R 4.0.2)                
#&gt;  rprojroot                1.3-2      2018-01-03 [2] RSPM (R 4.0.0)                
#&gt;  Rsamtools                2.5.3      2020-06-26 [1] Bioconductor                  
#&gt;  RSQLite                  2.2.1      2020-09-30 [1] RSPM (R 4.0.2)                
#&gt;  rstudioapi               0.11       2020-02-07 [2] RSPM (R 4.0.0)                
#&gt;  rtracklayer              1.49.5     2020-08-07 [1] Bioconductor                  
#&gt;  S4Vectors              * 0.27.14    2020-10-09 [1] Bioconductor                  
#&gt;  scales                   1.1.1      2020-05-11 [1] RSPM (R 4.0.0)                
#&gt;  sessioninfo            * 1.1.1      2018-11-05 [2] RSPM (R 4.0.0)                
#&gt;  shiny                    1.5.0      2020-06-23 [1] RSPM (R 4.0.2)                
#&gt;  stringi                  1.5.3      2020-09-09 [2] RSPM (R 4.0.2)                
#&gt;  stringr                  1.4.0      2019-02-10 [2] RSPM (R 4.0.0)                
#&gt;  SummarizedExperiment     1.19.9     2020-10-01 [1] Bioconductor                  
#&gt;  survival                 3.1-12     2020-04-10 [3] CRAN (R 4.0.2)                
#&gt;  systemfonts              0.3.2      2020-09-29 [1] RSPM (R 4.0.2)                
#&gt;  textshaping              0.1.2      2020-10-08 [1] RSPM (R 4.0.2)                
#&gt;  tibble                   3.0.3      2020-07-10 [2] RSPM (R 4.0.2)                
#&gt;  tidyselect               1.1.0      2020-05-11 [1] RSPM (R 4.0.0)                
#&gt;  VariantAnnotation        1.35.3     2020-06-26 [1] Bioconductor                  
#&gt;  vctrs                    0.3.4      2020-08-29 [2] RSPM (R 4.0.2)                
#&gt;  withr                    2.3.0      2020-09-22 [2] RSPM (R 4.0.2)                
#&gt;  xfun                     0.18       2020-09-29 [2] RSPM (R 4.0.2)                
#&gt;  XML                      3.99-0.5   2020-07-23 [1] RSPM (R 4.0.2)                
#&gt;  xml2                     1.3.2      2020-04-23 [2] RSPM (R 4.0.0)                
#&gt;  xtable                   1.8-4      2019-04-21 [1] RSPM (R 4.0.0)                
#&gt;  XVector                * 0.29.3     2020-06-25 [1] Bioconductor                  
#&gt;  yaml                     2.2.1      2020-02-01 [2] RSPM (R 4.0.0)                
#&gt;  zlibbioc                 1.35.0     2020-04-27 [1] Bioconductor                  
#&gt; 
#&gt; [1] /usr/local/lib/R/host-site-library
#&gt; [2] /usr/local/lib/R/site-library
#&gt; [3] /usr/local/lib/R/library</code></pre>
</div>
<div id="bibliography" class="section level1">
<h1 class="hasAnchor">
<a href="#bibliography" class="anchor"></a>Bibliography</h1>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.12/BiocStyle">BiocStyle</a></em> (<a href="https://github.com/Bioconductor/BiocStyle">Oleś, Morgan, and Huber, 2020</a>), <em><a href="https://CRAN.R-project.org/package=knitr">knitr</a></em> (<a href="http://www.crcpress.com/product/isbn/9781466561595">Xie, 2014</a>) and <em><a href="https://CRAN.R-project.org/package=rmarkdown">rmarkdown</a></em> (<a href="https://github.com/rstudio/rmarkdown">Allaire, Xie, McPherson, Luraschi, et al., 2020</a>) running behind the scenes.</p>
<p>Citations were made with <em><a href="https://CRAN.R-project.org/package=knitcitations">knitcitations</a></em> (<a href="https://github.com/cboettig/knitcitations">Boettiger, 2019</a>).</p>
<p>[1] J. Allaire, Y. Xie, J. McPherson, J. Luraschi, et al. <em>rmarkdown: Dynamic Documents for R</em>. R package version 2.4. 2020. &lt;URL: <a href="https://github.com/rstudio/rmarkdown" class="uri">https://github.com/rstudio/rmarkdown</a>&gt;.</p>
<p>[2] C. Boettiger. <em>knitcitations: Citations for ‘Knitr’ Markdown Files</em>. R package version 1.0.10. 2019. &lt;URL: <a href="https://github.com/cboettig/knitcitations" class="uri">https://github.com/cboettig/knitcitations</a>&gt;.</p>
<p>[3] J. Chávez, C. Barberena-Jonas, J. E. Sotelo-Fonseca, J. Alquicira-Hernandez, et al. “Programmatic access to bacterial regulatory networks with regutools”. In: <em>Bioinformatics</em> (2020). DOI: 10.1093/bioinformatics/btaa575. &lt;URL: <a href="https://academic.oup.com/bioinformatics/advance-article-abstract/doi/10.1093/bioinformatics/btaa575/5861528" class="uri">https://academic.oup.com/bioinformatics/advance-article-abstract/doi/10.1093/bioinformatics/btaa575/5861528</a>&gt;.</p>
<p>[4] J. Chávez, C. Barberena-Jonas, J. E. Sotelo-Fonseca, J. Alquicira-Hernandez, et al. <em>regutools: an R package for data extraction from RegulonDB</em>. <a href="https://github.com/comunidadbioinfo/regutools" class="uri">https://github.com/comunidadbioinfo/regutools</a> - R package version 1.1.3. 2020. DOI: 10.18129/B9.bioc.regutools. &lt;URL: <a href="http://www.bioconductor.org/packages/regutools" class="uri">http://www.bioconductor.org/packages/regutools</a>&gt;.</p>
<p>[5] G. Csárdi, R. core, H. Wickham, W. Chang, et al. <em>sessioninfo: R Session Information</em>. R package version 1.1.1. 2018. &lt;URL: <a href="https://github.com/r-lib/sessioninfo#readme" class="uri">https://github.com/r-lib/sessioninfo#readme</a>&gt;.</p>
<p>[6] Gustavsen, J. A., Pai, Shraddha, et al. “RCy3: Network Biology using Cytoscape from within R”. In: <em>F1000Research</em> (2019). DOI: 10.12688/f1000research.20887.3.</p>
<p>[7] F. Hahne and R. Ivanek. “Statistical Genomics: Methods and Protocols”. In: Ed. by E. Mathé and S. Davis. New York, NY: Springer New York, 2016. Chap. Visualizing Genomic Data Using Gviz and Bioconductor, pp. 335-351. ISBN: 978-1-4939-3578-9. DOI: 10.1007/978-1-4939-3578-9_16. &lt;URL: <a href="http://dx.doi.org/10.1007/978-1-4939-3578-9_16" class="uri">http://dx.doi.org/10.1007/978-1-4939-3578-9_16</a>&gt;.</p>
<p>[8] M. Lawrence, W. Huber, H. Pagès, P. Aboyoun, et al. “Software for Computing and Annotating Genomic Ranges”. In: <em>PLoS Computational Biology</em> 9 (8 2013). DOI: 10.1371/journal.pcbi.1003118. &lt;URL: <a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118" class="uri">http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118</a>}.&gt;</p>
<p>[9] M. Lawrence, W. Huber, H. Pagès, P. Aboyoun, et al. “Software for Computing and Annotating Genomic Ranges”. In: <em>PLoS Computational Biology</em> 9 (8 2013). DOI: 10.1371/journal.pcbi.1003118. &lt;URL: <a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118" class="uri">http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118</a>}.&gt;</p>
<p>[10] M. Morgan and L. Shepherd. <em>AnnotationHub: Client to access AnnotationHub resources</em>. R package version 2.21.5. 2020.</p>
<p>[11] K. Müller, H. Wickham, D. A. James, and S. Falcon. <em>RSQLite: ‘SQLite’ Interface for R</em>. <a href="https://rsqlite.r-dbi.org" class="uri">https://rsqlite.r-dbi.org</a>, <a href="https://github.com/r-dbi/RSQLite" class="uri">https://github.com/r-dbi/RSQLite</a>. 2020.</p>
<p>[12] A. Oleś, M. Morgan, and W. Huber. <em>BiocStyle: Standard styles for vignettes and other Bioconductor documents</em>. R package version 2.17.1. 2020. &lt;URL: <a href="https://github.com/Bioconductor/BiocStyle" class="uri">https://github.com/Bioconductor/BiocStyle</a>&gt;.</p>
<p>[13] H. Pagès, P. Aboyoun, R. Gentleman, and S. DebRoy. <em>Biostrings: Efficient manipulation of biological strings</em>. R package version 2.57.2. 2020.</p>
<p>[14] H. Pagès, M. Carlson, S. Falcon, and N. Li. <em>AnnotationDbi: Manipulation of SQLite-based annotations in Bioconductor</em>. R package version 1.51.3. 2020.</p>
<p>[15] H. Pagès, M. Lawrence, and P. Aboyoun. <em>S4Vectors: Foundation of vector-like and list-like containers in Bioconductor</em>. R package version 0.27.14. 2020.</p>
<p>[16] R Core Team. <em>R: A Language and Environment for Statistical Computing</em>. R Foundation for Statistical Computing. Vienna, Austria, 2020. &lt;URL: <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>&gt;.</p>
<p>[17] R Special Interest Group on Databases (R-SIG-DB), H. Wickham, and K. Müller. <em>DBI: R Database Interface</em>. <a href="http://r-dbi.github.io/DBI" class="uri">http://r-dbi.github.io/DBI</a>, <a href="https://github.com/r-dbi/DBI" class="uri">https://github.com/r-dbi/DBI</a>. 2019.</p>
<p>[18] L. Shepherd and M. Morgan. <em>BiocFileCache: Manage Files Across Sessions</em>. R package version 1.13.1. 2020.</p>
<p>[19] H. Wickham. “testthat: Get Started with Testing”. In: <em>The R Journal</em> 3 (2011), pp. 5-10. &lt;URL: <a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="uri">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>&gt;.</p>
<p>[20] Y. Xie. “knitr: A Comprehensive Tool for Reproducible Research in R”. In: <em>Implementing Reproducible Computational Research</em>. Ed. by V. Stodden, F. Leisch and R. D. Peng. ISBN 978-1466561595. Chapman and Hall/CRC, 2014. &lt;URL: <a href="http://www.crcpress.com/product/isbn/9781466561595" class="uri">http://www.crcpress.com/product/isbn/9781466561595</a>&gt;.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Joselyn Chavez, Carmina Barberena-Jonas, Jesus E. Sotelo-Fonseca, Leonardo Collado-Torres, Alejandro Reyes.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
